tests/basic_integration.rs

```diff
@@ -207,9 +207,33 @@ async fn test_pg_arrow_source() {
     assert!(!cdate_array.is_null(2));
     assert_eq!(cdate_array.value(2), elapsed_days as i32 + 2);
 
+    // THEN the first few numeric values should be as expected
+    let cnumeric_array = rb1
+        .column(11)
+        .as_any()
+        .downcast_ref::<Decimal128Array>()
+        .unwrap();
+    assert!(cnumeric_array.is_null(0));
+    assert!(!cnumeric_array.is_null(1));
+    assert_eq!(cnumeric_array.value(1), 0_i128);
+    assert!(!cnumeric_array.is_null(2));
+    assert_eq!(cnumeric_array.value(2), 1_i128);
+    assert!(!cnumeric_array.is_null(3));
+    assert_eq!(cnumeric_array.value(3), -2_i128);
+    assert!(!cnumeric_array.is_null(4));
+    assert_eq!(cnumeric_array.value(4), 3000_i128);
+    assert!(!cnumeric_array.is_null(5));
+    assert_eq!(cnumeric_array.value(5), -4000_i128);
+    assert!(!cnumeric_array.is_null(6));
+    assert_eq!(cnumeric_array.value(6), 50001_i128);
+    assert!(!cnumeric_array.is_null(7));
+    assert_eq!(cnumeric_array.value(7), 99999999_i128);
+    assert!(!cnumeric_array.is_null(8));
+    assert_eq!(cnumeric_array.value(8), -99999999_i128);
```

Comment:
Shouldn't this be `-99999.999` or is that the API of `.value()`? How do we check that it's properly scaled by 1e-3?

Reply:
try `assert_eq!(rb1.schema().field(11).data_type, Decimal256(8, 3))`?